File: lang/python/test/fun/vac/algorithms/search/LinearSearchTest.py
from fun.vac.algorithms.search import LinearSearch
from fun.vac.util import SequenceBuilder
from fun.vac.util import TestRunner

def testLinearSearch() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packIncreasing(arr)

    if LinearSearch.find(arr, -1) != size:
        return False

    if LinearSearch.find(arr, 2_147_483_647) != size:
        return False

    for i, v in enumerate(arr):
        if LinearSearch.find(arr, v) != i:
            return False

    return True

if __name__ == "__main__":
    TestRunner.parseTest(testLinearSearch())

File: lang/nodejs/test/fun/vac/algorithms/search/LinearSearchTest.js
"use strict"

const LinearSearch    = require("fun/vac/algorithms/search/LinearSearch")
const SequenceBuilder = require("fun/vac/util/SequenceBuilder")
const TestRunner      = require("fun/vac/util/TestRunner")

const testLinearSearch = () => {
    let size = 32768

    const arr = new Array(size)
    SequenceBuilder.packIncreasing(arr)

    if (LinearSearch.find(arr, -1) !== size) {
        return false
    }

    if (LinearSearch.find(arr, 2147483647) !== size) {
        return false
    }

    for (let i = 0; i < size; i++) {
        if (LinearSearch.find(arr, arr[i]) !== i) {
            return false
        }
    }

    return true
}

if (module === require.main) {
    TestRunner.parseTest(testLinearSearch())
}
